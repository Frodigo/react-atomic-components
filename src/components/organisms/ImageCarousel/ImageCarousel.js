import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';

import classes from './ImageCarousel.module.css';

const ImageCarousel = (props) => {
    const { items, title } = props;

    const carouselItems = items.map((item) => (
        <img
            key={`carousel-item-${item.alt}`}
            src={item.url}
            alt={item.alt}
            width="auto"
            height="auto"
        />
    ));

    const settings = {
        arrows: true,
        dots: false,
        infinite: true,
        speed: 500,
        variableWidth: false,
        slidesToShow: 4,
        responsive: [
            {
                breakpoint: 1023,
                settings: {
                    arrows: false,
                    variableWidth: true,
                    slidesToShow: 1
                }
            }
        ]
    };

    const shouldDisplayTitle = title ? (
        <h2 className={classes.title}>{title}</h2>
    ) : null;

    return (
        <div className={classes.carousel}>
            {shouldDisplayTitle}
            <Slider {...settings}>{carouselItems}</Slider>
        </div>
    );
};

ImageCarousel.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            alt: PropTypes.string,
            url: PropTypes.string
        })
    ).isRequired,
    title: PropTypes.string
};

export default ImageCarousel;
