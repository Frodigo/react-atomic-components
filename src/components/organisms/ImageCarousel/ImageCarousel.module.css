.carousel {
    padding: var(--spacing-md) 5px;
}

.title {
    padding: 0 10px;
    margin: var(--spacing-xl) 0 var(--spacing-xs);
}


@media (min-width: 1024px) {
    .carousel svg {
        width: 100%;
    }
}