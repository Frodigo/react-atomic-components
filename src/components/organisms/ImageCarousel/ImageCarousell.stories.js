import React from 'react';

import { default as ImageCarouselComponent } from './ImageCarousel';

export default {
    title: 'Organisms/ImageCarousel',
    component: ImageCarouselComponent
};

const items = [
    {
        url: 'https://via.placeholder.com/208x117',
        alt: 'figure'
    },
    {
        url: 'https://via.placeholder.com/260x117',
        alt: 'figure'
    },
    {
        url: 'https://via.placeholder.com/108x117',
        alt: 'figure'
    },
    {
        url: 'https://via.placeholder.com/58x117',
        alt: 'figure'
    }
];

const Template = (args) => <ImageCarouselComponent {...args} />;

export const ImageCarousel = Template.bind({});
ImageCarousel.args = {
    items,
    title: 'Image carousel'
};
