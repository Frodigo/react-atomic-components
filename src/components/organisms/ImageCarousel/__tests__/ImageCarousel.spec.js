import React from 'react';
import { render } from '@testing-library/react';

import ImageCarousel from '../ImageCarousel';

describe('Image carousel should be rendered', () => {
    const props = {
        items: [
            {
                url: 'https://via.placeholder.com/208x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/260x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/108x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/58x117',
                alt: 'figure'
            }
        ],
        title: 'Image carousel'
    };

    test('Image carousel renders correctly', () => {
        const { container } = render(<ImageCarousel {...props} />);

        expect(container).toMatchSnapshot();
    });
});
