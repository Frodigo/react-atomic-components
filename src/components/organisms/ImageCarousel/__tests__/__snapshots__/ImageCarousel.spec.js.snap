// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image carousel should be rendered Image carousel renders correctly 1`] = `
<div>
  <div
    class="carousel"
  >
    <h2
      class="title"
    >
      Image carousel
    </h2>
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="figure"
                height="auto"
                src="https://via.placeholder.com/208x117"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                width="auto"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="figure"
                height="auto"
                src="https://via.placeholder.com/260x117"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                width="auto"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="figure"
                height="auto"
                src="https://via.placeholder.com/108x117"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                width="auto"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="figure"
                height="auto"
                src="https://via.placeholder.com/58x117"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                width="auto"
              />
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;
